nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -211,6 +200,50 @@ public static IEnumerable<Dependency> GetTopLevelPackageDependenyInfos(Immutable
         }
     }
 
+    /// <summary>
+    /// Given an MSBuild string and a set of properties, returns our best guess at the final value MSBuild will evaluate to.
+    /// </summary>
+    /// <param name="msbuildString"></param>
+    /// <param name="propertyInfo"></param>
+    /// <returns></returns>
+    public static string GetRootValue(string msbuildString, Dictionary<string, string> propertyInfo)
+    {
+        bool modified;
+        do
+        {
+            modified = false;
+            if (HasPropertyReplace(msbuildString))
+            {
+                modified = false;
+                foreach (var property in propertyInfo)
```

Comment:
Should we just get the property name and perform the replacement instead of spinning through all the properties?

Reply:
We must spin through all the properties unfortunately because there could be nested property definition (`NewtonsoftJsonVersion`'s value is defined as the value of `NewtonsoftJsonV6Version` or some such).

Reply:
But we need to loop while property name is not null already, right?